src/resolve/resolved-dns-scope.c

```diff
@@ -306,55 +306,60 @@ static int dns_scope_socket(
                 int family,
                 const union in_addr_union *address,
                 DnsServer *server,
-                uint16_t port) {
+                uint16_t port,
+                union sockaddr_union **socket_address) {
 
         _cleanup_close_ int fd = -1;
-        union sockaddr_union sa = {};
+        _cleanup_free_ union sockaddr_union *sa;
         socklen_t salen;
         static const int one = 1;
         int r, ifindex;
 
         assert(s);
 
+        sa = new0(union sockaddr_union, 1);
```

Comment:
hmm, why do this with dynamic memory btw? I think the the callers should simply pass in a pre-allocated buffer, which this function fills on on success...

It's fine of we copy the structure around a bit... Today's CPUs excel at copying stuff around, and this structure isn't < 200 bytes iirc